### PR TITLE
Remove loop unroll

### PR DESCRIPTION
**Pull request by rhdxmr**
Since the Linux kernel v5.3, bounded loop has been supported.

Signed-off-by: Junyeong Jeong <rhdxmr@gmail.com>

### CODE REVIEW

**Comment by rhdxmr** [comment]
Cc @kevi-sun

**Comment by kevi-sun** [comment]
nice one!
I'm not quite sure whether someone might use old kernel version, to support that, sit might be good to have and option to toggle this action.

**Review by kevi-sun — APPROVED** [review]

**Comment by rsdy** [comment]
I agree, it seems sensible to put this behind a `feature` gate, may be on by default?

**Comment by rhdxmr** [comment]
Oh that sounds great!

Currently all loops are unrolled forcibly.
But I think it is not needed anymore since BPF supports bounded loops.

I think it is better to allow bounded loop by default and let users turn on the loop-unroll feature if they want.
Because the Linux kernel v5.3 is out of our build tests.
Now the oldest kernel version is v5.4 of ubuntu 20.04 LTS.

**Comment by rhdxmr** [comment]
Change cargo-bpf not to force loop unrolling.
`cargo bpf build --force-loop-unroll` option is added to enable full loop unrolling.

File: examples/example-userspace/examples/bounded-loop.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/// small. And you can see that the disassembled code is short and you can find
+/// bounded loop code in there. Since the Linux kernel v5.3, the bounded loop
+/// is permitted by the BPF verifier. So the default behavior of `cargo bpf
+/// build` is to preserve the bounded loop if the loop iteration count is big
+/// enough.
+///
+/// If `cargo bpf build --force-loop-unroll` is executed, the resulting BPF
+/// bytecode does not contain loop code. Instead, it contains fully unrolled
+/// loop. So you can see the disassembled code is pretty long.
+use std::path::PathBuf;
```
Comment (kevi-sun) [inline examples/example-userspace/examples/bounded-loop.rs:13]: good

**Review by kevi-sun — APPROVED** [review]

**Comment by rhdxmr** [comment]
@rsdy 
I added an option `--force-loop-unroll` to `cargo bpf` command.
what do you think about this?

**Review by rsdy — APPROVED** [review]

File: examples/example-userspace/examples/tc-map-share.rs [hunk 2]
```diff
@@ -27,7 +27,7 @@ async fn main() {
         .with_max_level(Level::TRACE)
         .finish();
     tracing::subscriber::set_global_default(subscriber).unwrap();
-    if unsafe { libc::getuid() != 0 } {
+    if unsafe { libc::geteuid() != 0 } {
```
Comment (rsdy) [inline examples/example-userspace/examples/tc-map-share.rs:30]: Just to note that the `geteuid()` changes are not strictly related to this patchset.

Comment (rhdxmr) [reply examples/example-userspace/examples/tc-map-share.rs:30]: That's true. It would be better this modification is sent via separate PR.
